### PR TITLE
[BugFix] improve lake primary table compaction strategy

### PR DESCRIPTION
**Pull request by luohaha**
Do some changes to lake primary table compaction policy:
1. Consider rowset's compaction score when picking rowsets.
2. When calc compaction score of tablet, only the picked rowsets will be involved.

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/storage/lake/compaction_policy.cpp [hunk 1]
```diff
@@ -154,16 +167,34 @@ StatusOr<std::vector<RowsetPtr>> PrimaryCompactionPolicy::pick_rowsets(int64_t v
         rowset_queue.pop();
     }
     VLOG(2) << strings::Substitute("lake PrimaryCompactionPolicy pick_rowsets tabletid:$0 version:$1 inputs:$2",
-                                   _tablet->id(), version, input_infos.str());
+                                   _tablet->id(), tablet_metadata->version(), input_infos.str());
 
     return input_rowsets;
 }
 
+Status primary_compaction_score_by_policy(const TabletMetadataPB& metadata, uint32_t& segment_num_score) {
```
Comment (wyb) [inline be/src/storage/lake/compaction_policy.cpp:175]:
```suggestion
StatusOr<uint32_t> primary_compaction_score_by_policy(const TabletMetadataPB& metadata) {
```

File: be/src/storage/lake/compaction_policy.cpp [hunk 2]
```diff
@@ -154,16 +167,34 @@ StatusOr<std::vector<RowsetPtr>> PrimaryCompactionPolicy::pick_rowsets(int64_t v
...
+}
+
 double primary_compaction_score(const TabletMetadataPB& metadata) {
     uint32_t segment_num_score = 0;
-    for (uint32_t i = 0; i < metadata.rowsets_size(); i++) {
-        const auto& rowset = metadata.rowsets(i);
-        segment_num_score += rowset.overlapped() ? rowset.segments_size() : 1;
+    // calc compaction score by picked rowsets
+    auto st = primary_compaction_score_by_policy(metadata, segment_num_score);
+    if (!st.ok()) {
```
Comment (wyb) [inline be/src/storage/lake/compaction_policy.cpp:190]: return 0 if failed?

Comment (luohaha) [reply be/src/storage/lake/compaction_policy.cpp]: Done

**Review by wyb — APPROVED** [review]

**Review by sduzh — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Pull request merged**